app/presenters/sufia/work_show_presenter.rb

```diff
@@ -42,6 +42,10 @@ def stats_path
       Sufia::Engine.routes.url_helpers.stats_work_path(self)
     end
 
+    def member_presenters(ids = ordered_ids, presenter_class = composite_presenter_class)
+      super.delete_if { |presenter| current_ability.cannot?(:read, presenter.solr_document) }
```

Comment:
This is a very slow implementation as it's going to do N+1 queries.  It should be possible to query it a single time with access controls.